File: Components/WinForms/CS/TreeListMainDemo/Modules/DragObject.cs
using System;

namespace DevExpress.XtraTreeList.Demos {
    //<imageListBoxControl1>
	public class DragObject {
		private string[] taskNames = new string[] {"Project", "Subtask", "Task"};
		private int index;
		public DragObject(int index) {
			this.index = index;
		}
		public object DragData { get { return new object[] { taskNames[index], DateTime.Now, DateTime.Now, 0} ; } }
		public int ImageIndex { get { return index; }}
	}
    //</imageListBoxControl1>
}

File: Components/WinForms/VB/TreeListMainDemo/Modules/DragObject.vb
Namespace DevExpress.XtraTreeList.Demos
	'<imageListBoxControl1>
	Public Class DragObject
		Private taskNames() As String = {"Project", "Subtask", "Task"}
		Private index As Integer
		Public Sub New(ByVal index As Integer)
			Me.index = index
		End Sub
		Public ReadOnly Property DragData() As Object
			Get
				Return New Object() { taskNames(index), Date.Now, Date.Now, 0}
			End Get
		End Property
		Public ReadOnly Property ImageIndex() As Integer
			Get
				Return index
			End Get
		End Property
	End Class
	'</imageListBoxControl1>
End Namespace
